python/paddle/fluid/reader.py

```diff
@@ -44,7 +47,354 @@ def _convert_places(places):
+            use_double_buffer (bool): whether to use double_buffer_reader to 
+                speed up data feeding. 
+            iterable (bool): whether the created reader object is iterable.
+            return_list (bool): whether the return value presented as list.
+
+        Returns:
+            loader (DataLoader): the created DataLoader object.
+
+        Examples:
+            
+            .. code-block:: python
+
+                import paddle.fluid as fluid
+                
+                image = fluid.layers.data(name='image', shape=[784], dtype='float32')
+                label = fluid.layers.data(name='label', shape=[1], dtype='int64')
+                loader = fluid.io.DataLoader.from_generator(feed_list=[image, label], capacity=16)
+        """
+        return GeneratorLoader(feed_list, capacity, use_double_buffer, iterable,
+                               return_list)
+
+    @staticmethod
+    def from_dataset(dataset, places, drop_last=True):
+        """
+        Create an iterable DataLoader object for loading data from Dataset.    
+        Dataset is only supported in Linux system currently.
+
+        Args:
+            dataset (InMemoryDataset|QueueDataset): the dataset object.
+            places (list(CUDAPlace)|list(CPUPlace)): places where the result data should be converted.   
+            drop_last (bool): whether to drop the last batch whose size is less than batch size. 
+
+        Returns:
+            loader (DataLoader): the created DataLoader object, which can be treated as a Python generator.   
+
+        Examples:
+
+            .. code-block:: python
+
+                import paddle.fluid as fluid
+
+                image = fluid.layers.data(name='image', shape=[784], dtype='float32')
+                label = fluid.layers.data(name='label', shape=[1], dtype='int64')
+
+                dataset = fluid.DatasetFactory().create_dataset("QueueDataset")
+                dataset.set_batch_size(32)
+                dataset.set_filelist(['a.txt', 'b.txt', 'c.txt'])
+                dataset.set_use_var([image, label])
+                dataset.set_pipe_command('cat') 
+
+                loader = fluid.io.DataLoader.from_dataset(dataset, fluid.cpu_places())
+        """
+        return DatasetLoader(dataset, places, drop_last)
+
+
+class GeneratorLoader(DataLoaderBase):
+    def __init__(self,
+                 feed_list=None,
+                 capacity=None,
+                 use_double_buffer=True,
+                 iterable=True,
+                 return_list=False):
+        self._tensor_reader = None
+        self._places = None
+        self._thread = None
+        self._feed_list = feed_list
+        if not capacity:
+            raise ValueError("Please give value to capacity.")
+        # force to use iterable mode under dygraph mode
+        if in_dygraph_mode():
+            if not iterable:
+                warnings.warn(
+                    "Please NOTE: dygraph can support iterable mode only.")
```

Comment:
Had better add "Change to iterable mode".

Reply:
Done.